pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +274,117 @@ func Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(in *v1.PodTemplateSpec,
 	return nil
 }
 
+// helper for v1.PodIPInfo list to core.PodIPinfoList
+func v1_PodIPs_To_core_PodIPs(inList []v1.PodIPInfo) []core.PodIPInfo {
+	out := make([]core.PodIPInfo, len(inList))
+	for idx, value := range inList {
+		out[idx] = core.PodIPInfo{
+			IP:         value.IP,
+			Properties: value.Properties,
+		}
+	}
+	return out
+}
+
+// Converts podStatus from v1 to core
+// auto + handling podIp/podIps
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	if err := autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// handle ip/ips fields
+	if 0 < len(in.PodIP) && 0 < len(in.PodIPs) {
+		// both fields are provided
+		if in.PodIP != in.PodIPs[0].IP {
+			return fmt.Errorf("conversion Error: in.PodIP(%v) != in.PodIPs[0](%v)", in.PodIP, in.PodIPs[0].IP)
+		}
+		out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
+	} else {
+		if 0 < len(in.PodIP) && 0 == len(in.PodIPs) {
+			// PodIP + NO entries in PodIPs
+			out.PodIPs = []core.PodIPInfo{
+				{
+					IP:         in.PodIP,
+					Properties: make(map[string]string),
+				},
+			}
+		} else {
+			// in.PodIP has no value
+			// copy the in.PodIPs
+			out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
+		}
+	}
+
+	// Verify no duplicates in core.PodIPs
+	// We compare IPs not the entire struct
+	ips := make([]core.PodIPInfo, 0)
+	for _, outer := range out.PodIPs {
+		bAdd := true
+		for _, inner := range ips {
+			if outer.IP == inner.IP {
+				bAdd = false
+				break
+			}
+		}
+
+		if bAdd {
+			ips = append(ips, outer)
+		}
+	}
+	out.PodIPs = ips
+
+	// PodFieldSelectors  generator depends on the fact that there is at least
+	// one ip
+	// check ./pkg/registry/core/pod/strategy.go
+	//TODO: (khenidak), change field selectors to work with multi ip
+	if 0 == len(out.PodIPs) {
+		out.PodIPs = []core.PodIPInfo{
+			{
+				IP: "",
```

Comment:
would probably be cleaner to conditionalize in PodToSelectableFields to treat `len(PodIPs)==0` as an IP of `""`

Reply:
i have a queued action item on it. This is a stop gap for now, just for me to be able to test the rest of the flows. I will fix before we merge.

Reply:
this is now fixed (i updated `PodToSelectableFields)`